include/core/shape.h

```diff
@@ -378,6 +378,178 @@ class Sphere : public Shape<dim>
 #endif
 };
 
+/**
+ * @class This class defines superquadric shapes. Their signed distance function
+ * is:
+ * |\frac{x}{a}|^r  + |\frac{y}{b}|^s + |\frac{z}{c}|^t - 1 = 0
+ * @tparam dim Dimension of the shape
+ */
+template <int dim>
+class Superquadric : public Shape<dim>
+{
+public:
+  /**
+   * @brief Constructor for a superquadric shape
+   * @param half_lengths The half-lengths of each direction
+   * @param exponents The blockiness in each direction
+   * @param position The superquadric center
+   * @param orientation The superquadric orientation
+   */
+  Superquadric<dim>(const Tensor<1, dim> half_lengths,
+                    const Tensor<1, dim> exponents,
+                    const Point<dim> &   position,
+                    const Tensor<1, 3> & orientation)
+    : Shape<dim>(half_lengths.norm(), position, orientation)
```

Comment:
```suggestion
    : Shape<dim>(half_lengths.norm(), position, orientation),half_lengths(half_lengths),exponents(exponents),epsilon(1e-12)
```
Is there a reason why half_lengths, exponents and epsilon are in the body of the constructor and not the initialization list ?